src/data/vocab_base.h

```diff
@@ -5,6 +5,25 @@
 #include "common/utils.h"
 #include "common/file_stream.h"
 
+#ifdef USE_SENTENCEPIECE
+#  if __GNUC__ >= 5
+#    pragma GCC diagnostic push
+#    pragma GCC diagnostic ignored "-Wsuggest-override"
+#  endif
+#  include "sentencepiece/src/sentencepiece_processor.h"
```

Comment:
So much for abstraction.  The generic vocab class shouldn't need to include sentencepiece.